pyccel/parser/semantic.py

```diff
@@ -102,33 +77,21 @@
 
 #==============================================================================
 
-from sympy.core.function       import Function, FunctionClass, Application
-from sympy.core.numbers        import ImaginaryUnit, IntegerConstant
-from sympy.logic.boolalg       import Boolean, BooleanTrue, BooleanFalse
+from sympy.core.function       import Application, UndefinedFunction
```

Comment:
Could you explain why moving from `Function` to `UndefinedFunction`? Thanks

Reply:
to be able to subclass it like this 
```
class UndefinedFunction(sp_UndefinedFunction, PyccelAstNode):
    pass
```
we can't do the same with Function

Reply:
Interesting.. thanks!